File: src/main/org/nlogo/api/Turtle3D.java
package org.nlogo.api ;

public interface Turtle3D extends Turtle
{
	/**
	 * Returns the value of the <code>pitch</code> variable
	 */
	double pitch() ;
	/**
	 * Returns the value of the <code>roll</code> variable
	 */
	double roll() ;
	/**
	 * Returns the value of the <code>zcor</code> variable
	 */
	double zcor() ;
	/**
	 * Returns the x component of the forward vector
	 */
	double dx() ;
	/**
	 * Returns the y component of the forward vector
	 */
	double dy() ;
	/**
	 * Returns the z component of the forward vector
	 */
	double dz() ;
}

File: src/main/org/nlogo/api/Turtle3D.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

trait Turtle3D extends Turtle {

  /** Returns the value of the <code>pitch</code> variable */
  def pitch: Double

  /** Returns the value of the <code>roll</code> variable */
  def roll: Double

  /** Returns the value of the <code>zcor</code> variable */
  def zcor: Double

  /** Returns the x component of the forward vector */
  def dx: Double

  /** Returns the y component of the forward vector */
  def dy: Double

  /** Returns the z component of the forward vector */
  def dz: Double

}
